csrc/transform_rfactor.cpp

```diff
@@ -185,11 +185,15 @@ class ReplayRFactor : public ReplayTransformations {
     Val* merged_id_size =
         mul(id_outer_mapped->extent(), id_inner_mapped->extent());
 
+    bool is_bcast =
+        id_outer_mapped->isBroadcast() && id_inner_mapped->isBroadcast();
+    auto iter_type = rfactor_axes_.count(m->out())
+        ? IterType::Reduction
+        : (is_bcast ? IterType::Broadcast : IterType::Iteration);
```

Comment:
naive question here, what's with `rfactor_axes_` here?

I know it's the old logic in existing code. I'm wondering if we should consider other possible `IterType` of the two operand?

Reply:
Rephrase my question. Are we making assumption of what `IterType` the merge operands have and should we assert that here?

Reply:
Good point, Let me try something similar to what we have in 
```
IterDomain* IterDomain::merge(
    IterDomain* outer,
    IterDomain* inner,
    bool rfactor_domain)
```
we have these iter types
```
// Symbolic: Undetermined between Iteration or Broadcast
enum class IterType {
  Iteration,
  Reduction,
  Broadcast,
  Gather,
  Stride,
  GatherScatter,
  VectorComponent,
  Symbolic
};
```

Reply:
(1) This is a replay of `IterDomain* IterDomain::merge()` and we have type checks in `IterDomain::merge()`, so there is no need to do another check.
(2) In `IterDomain::merge()`,  merging two  `GatherScatter` leads to a `GatherScatter`, but here we don't need to set IterType to `GatherScatter` since it is basically a special IterType and here we are not doing the actual gather_scatter, it is just a partial reduction.

Reply:
This fix seems to make sense, but ideally we should refactor this and also `handle(Split*)` such that they should just use `IterDomain::merge` and `IterDomain::split`, respectively. I don't know why that wasn't the case from the beginning, but that would reduce the duplication of the logic.

I remember we did fix something like this by changing some code to use `IterDomain::merge` and `IterDomain::split`. Maybe a different replay class, I suppose.

Reply:
Reasonable. Created an [issue](https://github.com/NVIDIA/Fuser/issues/2520) to track.